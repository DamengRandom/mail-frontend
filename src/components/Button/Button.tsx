import styled from "styled-components";

export type tButton = {
  type?: "button" | "submit" | "reset" | undefined,
  handleClick?: () => void,
  text: string,
  disabled?: boolean,
};

const ButtonWithStyle = styled.button`
  background-color: white;
  border-color: hsl(0, 0%, 80%);
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  outline: none;
  width: auto;
  cursor: pointer;
  padding: 10px;
  margin: 10px;
  font-family: Arial;
  font-weight: bold;
  
  &:hover {
    transition: all 1s;
    background-color: lightgreen;
    border-color: white;
  }
`;

export default function Button({ type, handleClick, text, disabled }: tButton) {
  return (
    <ButtonWithStyle
      onClick={handleClick}
      type={type}
      disabled={disabled}
    >{text}
    </ButtonWithStyle>
  );
};
