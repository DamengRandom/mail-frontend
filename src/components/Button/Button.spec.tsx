// import renderer from 'react-test-renderer';
import { cleanup, render, screen } from '@testing-library/react'
import Button, { tButton } from "./Button";

describe('<Button /> test', () => {
  let componentProps: JSX.IntrinsicAttributes & tButton,
      componentContainer: HTMLElement;

  beforeEach(() => {
    // Arrange
    const props: tButton = {
      type: "button",
      text: 'test text',
      disabled: false,
    };
    componentProps = props;
    // Act
    const { container } = render(<Button {...componentProps} />);
    componentContainer = container;
  });

  afterEach(cleanup);

  it('snapshot test for Button component', () => {
    // Assert
    expect(componentContainer.firstChild).toMatchSnapshot();
  });

  it('should render props for Button component', async () => {
    // Arrange
    componentProps = {
      ...componentProps,
      disabled: true,
    };
    // Act
    const { container } = render(<Button {...componentProps} />);
    const button = await screen.findAllByText('test text');
    // Assert
    expect(button).toBeDefined();
    expect((container.firstChild as HTMLInputElement).disabled).toBeTruthy();
  });
});
