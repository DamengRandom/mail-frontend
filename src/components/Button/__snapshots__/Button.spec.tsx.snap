// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> test snapshot test for Button component 1`] = `
<button
  class="sc-bdvvaa flvFCZ"
  type="button"
>
  test text
</button>
`;
